crates/stackable-operator/src/client.rs

```diff
@@ -622,7 +626,13 @@ where
     }
 }
 
-pub async fn create_client(field_manager: Option<String>) -> Result<Client> {
+pub async fn initialize_operator(field_manager: Option<String>) -> Result<Client> {
+    let _ = KUBERNETES_CLUSTER_DOMAIN
+        .set(retrieve_cluster_domain().context(ResolveKubernetesClusterDomainSnafu)?);
+    create_client(field_manager).await
+}
```

Comment:
Why is this a global?

Reply:
Its supposed to be the new "entry point" for operators. There will be more content in there in the future.

Reply:
It's basically renaming `create_client` to `initialize_operator` to be more expressive. By doing this we enforce that operators need to go throw this function and therefore initialize the OnceLock

Reply:
> It's basically renaming create_client to initialize_operator to be more expressive.

But it.. isn't?

Reply:
> There will be more content in there in the future.

Such as?

> By doing this we enforce that operators need to go throw this function and therefore initialize the OnceLock

This still doesn't explain *why* it needs to be a `OnceLock`.

Reply:
> Such as?

I have branch where I warn on an unsupported Kubernetes version. It's just not merged yet because of prioritization. In the future it may also read in the recommended product version or some global config object or whatnot or check if there is an update available.

> This still doesn't explain why it needs to be a OnceLock.

Agreed, I didn't try do explain that. Do you have an alternative suggestion how we can make this easy-to-use in operators? It looks very intuitive to me, but I'm not a Rust expert!
In the beginning we had a [LazyLock](https://doc.rust-lang.org/beta/std/sync/struct.LazyLock.html), but @Techassi mentioned that we would panic on e.g. an invalid DomainName being configured, and that the operators should error instead of panic, so we needed to change to a OnceLock.

Reply:
> I have branch where I warn on an unsupported Kubernetes version.

That still sounds like a part of.. creating a k8s client.

> Agreed, I didn't try do explain that. Do you have an alternative suggestion how we can make this easy-to-use in operators? It looks very intuitive to me, but I'm not a Rust expert!

I would have said you could store it in the `Client`, but on second thought maybe it would make more sense to create a separate `KubernetesClusterConfig` struct.

Reply:
We replaced the LacyLock with a KubernetesClusterInfo struct in https://github.com/stackabletech/operator-rs/pull/896